src/wallet/wallet.cpp

```diff
@@ -4069,23 +4069,53 @@ std::optional<MigrationData> CWallet::GetDescriptorsForLegacy(bilingual_str& err
 
     LegacyDataSPKM* legacy_spkm = GetLegacyDataSPKM();
     if (!Assume(legacy_spkm)) {
-        // This shouldn't happen
+        // This shouldn't happen, but we handle it gracefully.
```

Comment:
I'm not sure that this change preserves the intent of the error, could you explain why you came to the conclusion that this situation is not an internal bug in our handling of the wallet file?